resolve/index.go

```diff
@@ -97,10 +107,17 @@ type ruleRecord struct {
 //
 // kindToResolver is a map from rule kinds (for example, "go_library") to
 // Resolvers that support those kinds.
-func NewRuleIndex(mrslv func(r *rule.Rule, pkgRel string) Resolver) *RuleIndex {
+func NewRuleIndex(mrslv func(r *rule.Rule, pkgRel string) Resolver, exts ...interface{}) *RuleIndex {
+	var crossResolvers []CrossResolver
+	for _, e := range exts {
```

Comment:
Let's have callers pass in a slice of extensions rather than a slice of slices:

```
for _, e := range exts {
  if cr, ok := e.(CrossResolver); ok {
    crossResolvers = append(crossResolvers, cr)
  }
}
```